From 9ffdd233d5ef6b0d36277be04fe90a24a9cf53e1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Tue, 17 Apr 2012 16:41:19 +0200
Subject: [PATCH 02/19] progress dialog: add missing sanity check

---
 src/interface/dialog.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/interface/dialog.c b/src/interface/dialog.c
index f90e7c8..dcfb5dd 100644
--- a/src/interface/dialog.c
+++ b/src/interface/dialog.c
@@ -251,7 +251,11 @@ void dialog_ProgressDestroy (dialog_progress_bar_t *dialog)
 {
     assert (dialog);
 
-    dialog->pf_destroy (dialog->p_sys);
+    if (dialog->pf_destroy != NULL)
+    {
+        dialog->pf_destroy (dialog->p_sys);
+    }
+
     free (dialog);
 }
 
-- 
2.6.0

