From a4343a11f961b1f08f7a92aead30d0e96cacdad3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Tue, 17 Apr 2012 16:41:19 +0200
Subject: [PATCH 02/14] progress dialog: add missing sanity check

---
 src/interface/dialog.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/interface/dialog.c b/src/interface/dialog.c
index 3a0a36e..a9ac2e5 100644
--- a/src/interface/dialog.c
+++ b/src/interface/dialog.c
@@ -254,7 +254,11 @@ void dialog_ProgressDestroy (dialog_progress_bar_t *dialog)
 {
     assert (dialog);
 
-    dialog->pf_destroy (dialog->p_sys);
+    if (dialog->pf_destroy != NULL)
+    {
+        dialog->pf_destroy (dialog->p_sys);
+    }
+
     free (dialog);
 }
 
-- 
2.7.0

