From 3e6dab2ff7e8b459e065330a4de4c9c25d2f5d91 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Fri, 11 Sep 2015 18:34:26 +0200
Subject: [PATCH 08/16] libvlc: hack to disable daemon mode

---
 src/libvlc.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/libvlc.c b/src/libvlc.c
index 48c2b8c..3c9f879 100644
--- a/src/libvlc.c
+++ b/src/libvlc.c
@@ -200,6 +200,10 @@ int libvlc_InternalInit( libvlc_int_t *p_libvlc, int i_argc,
         return VLC_ENOMOD;
     }
 
+#ifdef __APPLE__
+#undef HAVE_DAEMON
+#endif
+
 #ifdef HAVE_DAEMON
     /* Check for daemon mode */
     if( var_InheritBool( p_libvlc, "daemon" ) )
-- 
2.9.2

