From 7c0963d6a837fc01e8ffc63f3066fcf601d6ac7f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Wed, 10 Dec 2014 22:14:55 +0100
Subject: [PATCH 08/14] disable neon volume plugin

---
 modules/arm_neon/Makefile.am | 1 -
 1 file changed, 1 deletion(-)

diff --git a/modules/arm_neon/Makefile.am b/modules/arm_neon/Makefile.am
index e67e07e..6909752 100644
--- a/modules/arm_neon/Makefile.am
+++ b/modules/arm_neon/Makefile.am
@@ -27,6 +27,5 @@ libyuv_rgb_neon_plugin_LIBTOOLFLAGS = --tag=CC
 if HAVE_NEON
 neon_LTLIBRARIES = \
 	libchroma_yuv_neon_plugin.la \
-	libvolume_neon_plugin.la \
 	libyuv_rgb_neon_plugin.la
 endif
-- 
2.7.0

