From 8c335f6a6ddfd0584c8dc804e03b37011b45fb10 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Fri, 11 Sep 2015 18:34:26 +0200
Subject: [PATCH 08/22] libvlc: hack to disable daemon mode

---
 src/libvlc.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/libvlc.c b/src/libvlc.c
index d3ce702..8cdc0bc 100644
--- a/src/libvlc.c
+++ b/src/libvlc.c
@@ -196,6 +196,10 @@ int libvlc_InternalInit( libvlc_int_t *p_libvlc, int i_argc,
         goto error;
     }
 
+#ifdef __APPLE__
+#undef HAVE_DAEMON
+#endif
+
 #ifdef HAVE_DAEMON
     /* Check for daemon mode */
     if( var_InheritBool( p_libvlc, "daemon" ) )
-- 
2.9.3

