From 01143837e290318cad524bf3bbae18ae1989706c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Thu, 10 Sep 2015 13:40:26 +0200
Subject: [PATCH 10/13] libvlc media list player: attach observer after player
 instance creation

---
 lib/media_list_player.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/media_list_player.c b/lib/media_list_player.c
index e6119ea..44f75a0 100644
--- a/lib/media_list_player.c
+++ b/lib/media_list_player.c
@@ -495,6 +495,7 @@ libvlc_media_list_player_new(libvlc_instance_t * p_instance)
         libvlc_event_manager_release(p_mlp->p_event_manager);
         goto error;
     }
+    install_media_player_observer(p_mlp);
 
     if (vlc_clone(&p_mlp->thread, playlist_thread, p_mlp,
                   VLC_THREAD_PRIORITY_LOW))
-- 
2.5.1

