From 83ab5bb6966e1484b2b3e80cf8708388b4b5929e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Fri, 12 Jun 2015 14:09:05 +0200
Subject: [PATCH] Sparkle: fix iOS compilation

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 0d1437b..49f4793 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3811,7 +3811,7 @@ dnl  MacOS X sparkle update support
 dnl
 AC_ARG_ENABLE(sparkle,
   [  --enable-sparkle        Sparkle update support for OS X (default enabled on Mac OS X)])
-if test "x${enable_sparkle}" != "xno" -a "${SYS}" = "darwin"
+if test "x${enable_sparkle}" != "xno" -a "${HAVE_OSX}" = "1"
 then
   if test ! -d ${CONTRIB_DIR}/Sparkle.framework
   then
-- 
2.4.3

