From ae6958130609926dd267d3315dea15504e7eba56 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Wed, 22 Jan 2014 13:43:11 +0100
Subject: [PATCH 10/19] contrib/avcodec: deploy libav

---
 contrib/src/ffmpeg/rules.mak | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/contrib/src/ffmpeg/rules.mak b/contrib/src/ffmpeg/rules.mak
index 5f73239..1b3fdad 100644
--- a/contrib/src/ffmpeg/rules.mak
+++ b/contrib/src/ffmpeg/rules.mak
@@ -1,7 +1,7 @@
 # FFmpeg
 
 #Uncomment the one you want
-#USE_LIBAV ?= 1
+USE_LIBAV ?= 1
 #USE_FFMPEG ?= 1
 
 ifdef USE_FFMPEG
-- 
1.8.5.2 (Apple Git-48)

