From a65807ff6051740d706f7c3a42f2a93f9b38eb3a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Wed, 22 Jan 2014 15:19:43 +0100
Subject: [PATCH 11/18] contrib/speexdsp: disable neon for AArch64

---
 contrib/src/speexdsp/rules.mak | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/contrib/src/speexdsp/rules.mak b/contrib/src/speexdsp/rules.mak
index e82c6ba..658a72e 100644
--- a/contrib/src/speexdsp/rules.mak
+++ b/contrib/src/speexdsp/rules.mak
@@ -31,6 +31,11 @@ ifeq ($(ARCH),arm)
 SPEEXDSP_CONF += --enable-arm5e-asm
 endif
 endif
+ifdef HAVE_IOS
+ifndef HAVE_ARMV7A
+SPEEXDSP_CONF += --disable-neon
+endif
+endif
 
 .speexdsp: speexdsp
 	cd $< && $(HOSTVARS) ./configure $(HOSTCONF) $(SPEEXDSP_CONF)
-- 
2.4.3

