From 1ee8245688b61fcd5fcdb81973faac5e8db70a6b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Thu, 3 Oct 2013 18:28:26 +0200
Subject: [PATCH 11/19] extras: deploy arch option for iOS

---
 extras/package/ios/build.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/extras/package/ios/build.sh b/extras/package/ios/build.sh
index 32536ea..972517c 100755
--- a/extras/package/ios/build.sh
+++ b/extras/package/ios/build.sh
@@ -187,7 +187,7 @@ else
     export ASCPP="xcrun as"
 fi
 
-../bootstrap --host=x86_64-apple-darwin11 --build=${TARGET} --prefix=${VLCROOT}/contrib/${TARGET}-${ARCH} --disable-gpl \
+../bootstrap --host=x86_64-apple-darwin11 --build=${TARGET} --arch=${ARCH} --prefix=${VLCROOT}/contrib/${TARGET}-${ARCH} --disable-gpl \
     --disable-disc --disable-sout \
     --disable-sdl \
     --disable-SDL_image \
-- 
1.8.3.4 (Apple Git-47)

