From 485b73abcc8ed246c697083b209691ca6fa5c1a9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Mon, 26 Oct 2015 10:47:56 +0100
Subject: [PATCH 12/16] disable videotoolbox

---
 extras/package/ios/build.sh | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/extras/package/ios/build.sh b/extras/package/ios/build.sh
index 6789738..c480171 100755
--- a/extras/package/ios/build.sh
+++ b/extras/package/ios/build.sh
@@ -486,6 +486,8 @@ output_http
 output_livehttp
 libmux
 stream_out
+videotoolbox
+cvpx_i420
 "
 
 if [ "$SCARY" = "no" ]; then
-- 
2.7.2

