From 407b787ae87531dc69baf3399cc566000e2bf415 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Wed, 22 Jan 2014 15:19:43 +0100
Subject: [PATCH 14/19] contrib/speexdsp: disable neon for AArch64

---
 contrib/src/speexdsp/rules.mak | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/contrib/src/speexdsp/rules.mak b/contrib/src/speexdsp/rules.mak
index 3cc2b15..049565d 100644
--- a/contrib/src/speexdsp/rules.mak
+++ b/contrib/src/speexdsp/rules.mak
@@ -36,6 +36,11 @@ ifeq ($(ARCH),arm)
 SPEEXDSP_CONF += --enable-arm5e-asm
 endif
 endif
+ifdef HAVE_IOS
+ifndef HAVE_ARMV7A
+SPEEXDSP_CONF += --disable-neon
+endif
+endif
 
 .speexdsp: speexdsp
 	mkdir -p $</m4 && $(RECONF)
-- 
1.9.3 (Apple Git-50)

