From f087abc2ac103eabccc324b4dea519bf8a51f5eb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Wed, 22 Jan 2014 13:37:59 +0100
Subject: [PATCH 15/18] contrib/speex: fix compilation for non-ARMv7 iOS
 targets

---
 contrib/src/speex/rules.mak | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/contrib/src/speex/rules.mak b/contrib/src/speex/rules.mak
index a5f003c..916b0ea 100644
--- a/contrib/src/speex/rules.mak
+++ b/contrib/src/speex/rules.mak
@@ -27,6 +27,11 @@ endif
 ifeq ($(ARCH),aarch64)
 SPEEX_CONF += --disable-neon
 endif
+ifdef HAVE_IOS
+ifndef HAVE_ARMV7A
+SPEEX_CONF += --disable-neon
+endif
+endif
 
 .speex: speex
 	cd $< && $(HOSTVARS) ./configure $(HOSTCONF) $(SPEEX_CONF)
-- 
2.4.3

