From 91c85d1b2e73d0d67aba16ae30f1f5511e6cb78c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Wed, 22 Jan 2014 15:19:43 +0100
Subject: [PATCH 15/20] contrib/speexdsp: disable neon for AArch64

---
 contrib/src/speexdsp/rules.mak | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/contrib/src/speexdsp/rules.mak b/contrib/src/speexdsp/rules.mak
index 7a2bc80..396af52 100644
--- a/contrib/src/speexdsp/rules.mak
+++ b/contrib/src/speexdsp/rules.mak
@@ -29,6 +29,11 @@ ifeq ($(ARCH),arm)
 SPEEXDSP_CONF += --enable-arm5e-asm
 endif
 endif
+ifdef HAVE_IOS
+ifndef HAVE_ARMV7A
+SPEEXDSP_CONF += --disable-neon
+endif
+endif
 
 .speexdsp: speexdsp
 	mkdir -p $</m4 && $(RECONF)
-- 
1.8.5.2 (Apple Git-48)

