From 5814dff5bf72cf9db81a6aed15bd88f640879812 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Wed, 15 Jan 2014 15:26:46 +0100
Subject: [PATCH 16/19] contrib/opus disable asm on arm

---
 contrib/src/opus/rules.mak | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/contrib/src/opus/rules.mak b/contrib/src/opus/rules.mak
index d783648..f09a112 100644
--- a/contrib/src/opus/rules.mak
+++ b/contrib/src/opus/rules.mak
@@ -23,6 +23,9 @@ OPUS_CONF=
 ifndef HAVE_FPU
 OPUS_CONF += --enable-fixed-point
 endif
+ifdef HAVE_IOS
+OPUS_CONF += --disable-asm
+endif
 
 .opus: opus
 	cd $< && $(HOSTVARS) ./configure $(HOSTCONF) $(OPUS_CONF)
-- 
1.8.3.4 (Apple Git-47)

