From 89f0df4cf8d4b1dc3b8f65558fb27dd7db616208 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Mon, 23 Nov 2015 15:51:10 +0100
Subject: [PATCH 16/21] freetype: enable font fallback on Darwin

---
 modules/text_renderer/text_layout.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/modules/text_renderer/text_layout.c b/modules/text_renderer/text_layout.c
index 73615ce..b0c840d 100644
--- a/modules/text_renderer/text_layout.c
+++ b/modules/text_renderer/text_layout.c
@@ -89,6 +89,11 @@
 # define HAVE_FONT_FALLBACK
 #endif
 
+/* Darwin */
+#ifdef __APPLE__
+# define HAVE_FONT_FALLBACK
+#endif
+
 /**
  * Within a paragraph, run_desc_t represents a run of characters
  * having the same font face, size, and style, Unicode script
-- 
2.6.0

