From 38f6edd3578019a7e55e9ed5d2985d18ac62a8b0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Wed, 15 Jan 2014 15:15:12 +0100
Subject: [PATCH 16/21] contrib: don't try to guess ARCH if it was given as an
 argument

---
 contrib/src/main.mak | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/contrib/src/main.mak b/contrib/src/main.mak
index 23a47e9..352e557 100644
--- a/contrib/src/main.mak
+++ b/contrib/src/main.mak
@@ -35,7 +35,9 @@ PREFIX := $(abspath $(PREFIX))
 ifneq ($(HOST),$(BUILD))
 HAVE_CROSS_COMPILE = 1
 endif
+ifndef ARCH
 ARCH := $(shell $(SRC)/get-arch.sh $(HOST))
+endif
 
 ifeq ($(ARCH)-$(HAVE_WIN32),x86_64-1)
 HAVE_WIN64 := 1
-- 
1.8.3.4 (Apple Git-47)

