From 9f87bb57ccea56615104c03da503b965e4e97c1d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Mon, 28 Dec 2015 15:10:18 +0100
Subject: [PATCH 17/17] HACK to temporarily fix TS seeking

---
 modules/demux/mpeg/ts.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/modules/demux/mpeg/ts.c b/modules/demux/mpeg/ts.c
index ac739bd..e0e8c2f 100644
--- a/modules/demux/mpeg/ts.c
+++ b/modules/demux/mpeg/ts.c
@@ -1523,8 +1523,8 @@ static int Control( demux_t *p_demux, int i_query, va_list args )
                  i_length > 0 && !SeekToTime( p_demux, p_pmt, TO_SCALE(i_length) * f ) )
             {
                 ReadyQueuesPostSeek( p_demux );
-                es_out_Control( p_demux->out, ES_OUT_SET_NEXT_DISPLAY_TIME,
-                                TO_SCALE(i_length) * f );
+                /*es_out_Control( p_demux->out, ES_OUT_SET_NEXT_DISPLAY_TIME,
+                                TO_SCALE(i_length) * f );*/
                 return VLC_SUCCESS;
             }
         }
@@ -1539,8 +1539,8 @@ static int Control( demux_t *p_demux, int i_query, va_list args )
             if( !SeekToTime( p_demux, p_pmt, p_pmt->pcr.i_first + i_length * f ) )
             {
                 ReadyQueuesPostSeek( p_demux );
-                es_out_Control( p_demux->out, ES_OUT_SET_NEXT_DISPLAY_TIME,
-                                FROM_SCALE(p_pmt->pcr.i_first + i_length * f) );
+                /*es_out_Control( p_demux->out, ES_OUT_SET_NEXT_DISPLAY_TIME,
+                                FROM_SCALE(p_pmt->pcr.i_first + i_length * f) );*/
                 return VLC_SUCCESS;
             }
         }
-- 
2.6.4

