From 1e53afaed4d52fb9894d4e47cd30266eae816f61 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Mon, 26 Oct 2015 11:38:11 +0100
Subject: [PATCH 17/18] contrib/freetype: fix compilation on Darwin

---
 contrib/src/freetype2/rules.mak | 1 +
 1 file changed, 1 insertion(+)

diff --git a/contrib/src/freetype2/rules.mak b/contrib/src/freetype2/rules.mak
index 4b57ff3..d001e9b 100644
--- a/contrib/src/freetype2/rules.mak
+++ b/contrib/src/freetype2/rules.mak
@@ -17,6 +17,7 @@ freetype: freetype-$(FREETYPE2_VERSION).tar.gz .sum-freetype2
 	$(UNPACK)
 	$(call pkg_static, "builds/unix/freetype2.in")
 	$(MOVE)
+	cd $@ && cp builds/unix/install-sh .
 
 DEPS_freetype2 = zlib $(DEPS_zlib)
 
-- 
2.6.1

