From 89303da8dcc3c6e46444f2adbd4c53ff6a662b7f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Sun, 15 Nov 2015 01:30:56 +0100
Subject: [PATCH 18/20] extras/tools: update ant

---
 extras/tools/packages.mak | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/extras/tools/packages.mak b/extras/tools/packages.mak
index 175548c..4590a73 100644
--- a/extras/tools/packages.mak
+++ b/extras/tools/packages.mak
@@ -41,7 +41,7 @@ RAGEL_URL=http://www.colm.net/files/ragel/ragel-$(RAGEL_VERSION).tar.gz
 SED_VERSION=4.2.2
 SED_URL=$(GNU)/sed/sed-$(SED_VERSION).tar.bz2
 
-ANT_VERSION=1.9.5
+ANT_VERSION=1.9.6
 ANT_URL=$(APACHE)/ant/binaries/apache-ant-$(ANT_VERSION)-bin.tar.bz2
 
 PROTOBUF_VERSION := 2.6.1
-- 
2.6.3

