From 6f0773f75024689466af987f179438642d465f13 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Tue, 12 Jan 2016 19:42:30 +0100
Subject: [PATCH 18/18] securetransport: fix segfault

---
 modules/misc/securetransport.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/modules/misc/securetransport.c b/modules/misc/securetransport.c
index 387d2cd..3aac855 100644
--- a/modules/misc/securetransport.c
+++ b/modules/misc/securetransport.c
@@ -386,7 +386,9 @@ static int st_Handshake (vlc_tls_creds_t *crd, vlc_tls_t *session,
     vlc_tls_sys_t *sys = session->sys;
 
     OSStatus retValue = SSLHandshake(sys->p_context);
-    *alp = NULL;
+    if (alp != NULL) {
+        *alp = NULL;
+    }
 
     if (retValue == errSSLWouldBlock) {
         msg_Dbg(crd, "handshake is blocked, try again later");
-- 
2.7.0

