From a48e9b134a145ab6702b20d21876ea6e3bbb5250 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Wed, 22 Jan 2014 12:21:12 +0100
Subject: [PATCH 18/21] contrib: fix iOS simulator architecture features

---
 contrib/bootstrap | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/contrib/bootstrap b/contrib/bootstrap
index 5e5f935..6d5753b 100755
--- a/contrib/bootstrap
+++ b/contrib/bootstrap
@@ -219,7 +219,13 @@ case "${OS}" in
 			add_make_enabled "HAVE_MACOSX" "HAVE_DARWIN_OS" "HAVE_BSD"
 		else
 			check_ios_sdk
-			add_make_enabled "HAVE_IOS" "HAVE_DARWIN_OS" "HAVE_BSD" "HAVE_NEON" "HAVE_ARMV7A"
+ 			add_make_enabled "HAVE_IOS" "HAVE_DARWIN_OS" "HAVE_BSD"
+ 			if [ "$ARCH" != "x86_64" && "$ARCH" != "i386" ]; then
+ 				add_make_enabled "HAVE_NEON"
+ 				if [ "$ARCH" != "arm64" ]; then
+ 				add_make_enabled "HAVE_ARMV7A"
+ 				fi
+ 			fi
 		fi
 		;;
 	*bsd*)
-- 
1.8.3.4 (Apple Git-47)

