From 462d2be9d8af203977ff5ebd17fcb754a5cc5810 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Tue, 31 Dec 2013 14:45:31 +0100
Subject: [PATCH] contrib/avcodec: use the other side

---
 contrib/src/ffmpeg/rules.mak | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/contrib/src/ffmpeg/rules.mak b/contrib/src/ffmpeg/rules.mak
index d908ca3..1b41466 100644
--- a/contrib/src/ffmpeg/rules.mak
+++ b/contrib/src/ffmpeg/rules.mak
@@ -1,8 +1,8 @@
 # FFmpeg
 
 #Uncomment the one you want
-#USE_LIBAV ?= 1
-USE_FFMPEG ?= 1
+USE_LIBAV ?= 1
+#USE_FFMPEG ?= 1
 
 ifdef USE_FFMPEG
 HASH=74a9c92840d3494393938874c1cdd5bdb95cc918
-- 
1.8.3.4 (Apple Git-47)

