From 438bcbf0cfeffd99667fb477a42973e25ec9a8f9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Wed, 15 Jan 2014 15:13:03 +0100
Subject: [PATCH 20/24] tools: update gas-pp

---
 extras/tools/packages.mak | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/extras/tools/packages.mak b/extras/tools/packages.mak
index 085f0ef..a4ce1db 100644
--- a/extras/tools/packages.mak
+++ b/extras/tools/packages.mak
@@ -32,7 +32,7 @@ XZ_URL=http://tukaani.org/xz/xz-$(XZ_VERSION).tar.bz2
 OPENSSL_VERSION=1.0.0e
 OPENSSL_URL=http://www.openssl.org/source/openssl-$(OPENSSL_VERSION).tar.gz
 
-GAS_VERSION=35de474
+GAS_VERSION=940274b
 GAS_URL=http://git.libav.org/?p=gas-preprocessor.git;a=snapshot;h=$(GAS_VERSION);sf=tgz
 
 RAGEL_VERSION=6.8
-- 
1.8.3.4 (Apple Git-47)

