From 1d0384598544fe52176003898918b595cd344909 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Wed, 22 Jan 2014 12:20:52 +0100
Subject: [PATCH 28/29] contrib/png: fix asm compilation for ARMv7

---
 contrib/src/png/rules.mak | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/contrib/src/png/rules.mak b/contrib/src/png/rules.mak
index d76206e..4f12595 100644
--- a/contrib/src/png/rules.mak
+++ b/contrib/src/png/rules.mak
@@ -9,7 +9,9 @@ endif
 
 PNGCONF =
 ifdef HAVE_IOS
-ifeq ($(ARCH),arm64)
+ifdef HAVE_ARMV7A
+PNGCONF = --enable-arm-neon=yes
+else
 PNGCONF = --enable-arm-neon=no
 endif
 endif
-- 
1.8.3.4 (Apple Git-47)

