From 898a4180bddfd1c1517b96f191c30457d4178324 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <felix@feepk.net>
Date: Mon, 15 Feb 2021 13:44:41 +0100
Subject: [PATCH 07/14] contrib: disable rust when compiling on macOS ARM-64

---
 contrib/src/main-rust.mak | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/contrib/src/main-rust.mak b/contrib/src/main-rust.mak
index 13025ec17d..3fbc147b49 100644
--- a/contrib/src/main-rust.mak
+++ b/contrib/src/main-rust.mak
@@ -24,9 +24,11 @@ endif
 endif
 endif
 else ifdef HAVE_MACOSX
-ifneq ($(ARCH),aarch64) # macOS ARM-64 is unsupported
+ifneq ($(ARCH),aarch64) # running on macOS ARM-64 is unsupported
+ifeq ($(BUILD),x86_64-apple-darwin) # building on macOS ARM-64 is unsupported
 RUST_TARGET = $(ARCH)-apple-darwin
 endif
+endif
 else ifdef HAVE_SOLARIS
 RUST_TARGET = x86_64-sun-solaris
 else ifdef HAVE_LINUX
-- 
2.32.1 (Apple Git-133)

