From 71bc0c3fea418826c42ab3294d9df956fb3a1ff2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <felix@feepk.net>
Date: Mon, 26 Oct 2020 15:47:13 +0100
Subject: [PATCH 08/20] extras: disable rav1e for iOS and tvOS

---
 extras/package/apple/build.conf | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/extras/package/apple/build.conf b/extras/package/apple/build.conf
index 7b51b16504..e24903ab8e 100644
--- a/extras/package/apple/build.conf
+++ b/extras/package/apple/build.conf
@@ -85,12 +85,14 @@ export VLC_CONTRIB_OPTIONS_MACOSX=(
 # Additional contrib bootstrap options for iOS
 export VLC_CONTRIB_OPTIONS_IOS=(
     --disable-aom
+    --disable-rav1e
 )
 
 # Additional contrib bootstrap options for tvOS
 export VLC_CONTRIB_OPTIONS_TVOS=(
     --disable-libarchive
     --disable-aom
+    --disable-rav1e
 )
 
 #
-- 
2.32.0 (Apple Git-132)

