From 864225d55413e016b43d1c8581012217f94ac29f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <felix@feepk.net>
Date: Tue, 16 Feb 2021 17:04:59 +0100
Subject: [PATCH 10/18] contrib/ffmpeg: fix cross-compilation for macOS on
 macOS

---
 contrib/src/ffmpeg/rules.mak | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/contrib/src/ffmpeg/rules.mak b/contrib/src/ffmpeg/rules.mak
index 0aa82fdeac..309b1edc0e 100644
--- a/contrib/src/ffmpeg/rules.mak
+++ b/contrib/src/ffmpeg/rules.mak
@@ -139,6 +139,8 @@ FFMPEGCONF += --enable-pic --extra-ldflags="$(EXTRA_CFLAGS) -isysroot $(IOS_SDK)
 ifdef HAVE_NEON
 FFMPEGCONF += --as="$(AS)"
 endif
+else
+FFMPEGCONF += --enable-pic --extra-ldflags="$(EXTRA_CFLAGS) -isysroot $(MACOSX_SDK)"
 endif
 endif
 
-- 
2.32.1 (Apple Git-133)

