From d68ed9c68c1fb7ba75de4bfc7541f27ed5c3cabc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <felix@feepk.net>
Date: Tue, 16 Feb 2021 17:04:59 +0100
Subject: [PATCH 11/20] contrib/ffmpeg: fix cross-compilation for macOS on
 macOS

---
 contrib/src/ffmpeg/rules.mak | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/contrib/src/ffmpeg/rules.mak b/contrib/src/ffmpeg/rules.mak
index 1dcb2b4b9c..8a1951befa 100644
--- a/contrib/src/ffmpeg/rules.mak
+++ b/contrib/src/ffmpeg/rules.mak
@@ -139,6 +139,8 @@ FFMPEGCONF += --enable-pic --extra-ldflags="$(EXTRA_CFLAGS) -isysroot $(IOS_SDK)
 ifdef HAVE_NEON
 FFMPEGCONF += --as="$(AS)"
 endif
+else
+FFMPEGCONF += --enable-pic --extra-ldflags="$(EXTRA_CFLAGS) -isysroot $(MACOSX_SDK)"
 endif
 endif
 
-- 
2.32.0 (Apple Git-132)

