From 6bc6483a7dfa763080d73b3d5b7ea2a16b70fc2f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <felix@feepk.net>
Date: Mon, 21 Apr 2025 19:35:58 +0200
Subject: [PATCH 12/12] apple/build.conf: disable ytdl

---
 extras/package/apple/build.conf | 1 +
 1 file changed, 1 insertion(+)

diff --git a/extras/package/apple/build.conf b/extras/package/apple/build.conf
index 116529fa7b..86af8564ad 100644
--- a/extras/package/apple/build.conf
+++ b/extras/package/apple/build.conf
@@ -249,6 +249,7 @@ export VLC_MODULE_REMOVAL_LIST_BASE=(
     fingerprinter
     access_output_udp
     access_output_livehttp
+    ytdl
 )
 
 # Additional modules to remove for macOS
-- 
2.39.5 (Apple Git-154)

