From e50f703c3ee9dc70eb3ce30fdebd7e03db933588 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <felix@feepk.net>
Date: Tue, 16 Feb 2021 17:04:59 +0100
Subject: [PATCH 14/14] contrib/ffmpeg: fix cross-compilation for macOS on
 macOS

---
 contrib/src/ffmpeg/rules.mak | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/contrib/src/ffmpeg/rules.mak b/contrib/src/ffmpeg/rules.mak
index 2cac72934c..22ad2277ec 100644
--- a/contrib/src/ffmpeg/rules.mak
+++ b/contrib/src/ffmpeg/rules.mak
@@ -143,6 +143,8 @@ FFMPEGCONF += --cpu=core2
 endif
 ifdef HAVE_IOS
 FFMPEGCONF += --enable-pic --extra-ldflags="$(EXTRA_CFLAGS) -isysroot $(IOS_SDK)"
+else
+FFMPEGCONF += --enable-pic --extra-ldflags="$(EXTRA_CFLAGS) -isysroot $(MACOSX_SDK)"
 endif
 endif
 
-- 
2.39.2 (Apple Git-143)

