From 9d05d4c2a240f30f223b59e95fb583bb7a87635b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <felix@feepk.net>
Date: Sun, 17 Mar 2024 14:35:16 +0100
Subject: [PATCH 18/18] videotoolbox: fix compilation and renderering on xrOS

---
 modules/codec/videotoolbox/decoder.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/modules/codec/videotoolbox/decoder.c b/modules/codec/videotoolbox/decoder.c
index 2e5316853b..4282fcd437 100644
--- a/modules/codec/videotoolbox/decoder.c
+++ b/modules/codec/videotoolbox/decoder.c
@@ -1178,9 +1178,15 @@ static int StartVideoToolbox(decoder_t *p_dec)
                          kCVPixelBufferIOSurfaceOpenGLTextureCompatibilityKey,
                          kCFBooleanTrue);
 #else
+#ifndef TARGET_OS_VISION
     CFDictionarySetValue(destinationPixelBufferAttributes,
                          kCVPixelBufferOpenGLESCompatibilityKey,
                          kCFBooleanTrue);
+#else
+    CFDictionarySetValue(destinationPixelBufferAttributes,
+                         kCVPixelBufferMetalCompatibilityKey,
+                         kCFBooleanTrue);
+#endif
 #endif
 
     cfdict_set_int32(destinationPixelBufferAttributes,
-- 
2.39.3 (Apple Git-146)

