From 17e2fda214de9c912b876a5bc59c3f879ef1e7e2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <felix@feepk.net>
Date: Wed, 22 Jan 2020 17:59:47 +0100
Subject: [PATCH 18/18] contrib/gpg-error: remove appleOS hacks

---
 contrib/src/gpg-error/rules.mak | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/contrib/src/gpg-error/rules.mak b/contrib/src/gpg-error/rules.mak
index badf00cf57..d08ac94384 100644
--- a/contrib/src/gpg-error/rules.mak
+++ b/contrib/src/gpg-error/rules.mak
@@ -32,19 +32,6 @@ else
 	cp $@/src/syscfg/lock-obj-pub.arm-unknown-linux-androideabi.h $@/src/syscfg/lock-obj-pub.linux-android.h
 endif
 endif
-ifdef HAVE_DARWIN_OS
-ifdef HAVE_ARMV7A
-	cp $@/src/syscfg/lock-obj-pub.arm-apple-darwin.h $@/src/syscfg/lock-obj-pub.$(HOST).h
-else
-ifeq ($(ARCH),aarch64)
-ifneq ($(HOST),aarch64-apple-darwin)
-	cp $@/src/syscfg/lock-obj-pub.aarch64-apple-darwin.h $@/src/syscfg/lock-obj-pub.$(HOST).h
-endif
-else
-	cp $@/src/syscfg/lock-obj-pub.x86_64-apple-darwin.h $@/src/syscfg/lock-obj-pub.$(HOST).h
-endif
-endif
-endif
 ifdef HAVE_NACL
 ifeq ($(ARCH),i386) # 32bits intel
 	cp $@/src/syscfg/lock-obj-pub.i686-pc-linux-gnu.h $@/src/syscfg/lock-obj-pub.nacl.h
-- 
2.21.1 (Apple Git-122.3)

