From e810d37d8f25838af1b25cb52970c1d4745e6860 Mon Sep 17 00:00:00 2001
From: Benjamin King <king7532@mac.com>
Date: Sat, 7 Oct 2023 17:34:41 -0500
Subject: [PATCH 18/19] codec/videotoolbox: Add support for visionOS

---
 modules/codec/videotoolbox/decoder.c | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/modules/codec/videotoolbox/decoder.c b/modules/codec/videotoolbox/decoder.c
index 2e5316853b..2110b6296c 100644
--- a/modules/codec/videotoolbox/decoder.c
+++ b/modules/codec/videotoolbox/decoder.c
@@ -1173,11 +1173,15 @@ static int StartVideoToolbox(decoder_t *p_dec)
         goto error;
     }
 
-#if !TARGET_OS_IPHONE
+#if TARGET_OS_OSX
+   CFDictionarySetValue(destinationPixelBufferAttributes,
+                        kCVPixelBufferIOSurfaceOpenGLTextureCompatibilityKey,
+                        kCFBooleanTrue);
+#elif defined(TARGET_OS_VISION) && TARGET_OS_VISION
     CFDictionarySetValue(destinationPixelBufferAttributes,
-                         kCVPixelBufferIOSurfaceOpenGLTextureCompatibilityKey,
+                         kCVPixelBufferMetalCompatibilityKey,
                          kCFBooleanTrue);
-#else
+#else // iOS and tvOS
     CFDictionarySetValue(destinationPixelBufferAttributes,
                          kCVPixelBufferOpenGLESCompatibilityKey,
                          kCFBooleanTrue);
-- 
2.32.1 (Apple Git-133)

