From d9ff2170cc244d26ecc83b076698b556f3043d83 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Fri, 31 Jul 2026 15:10:32 +0200
Subject: [PATCH 19/27] playlist/podcast: expose description on input item

---
 modules/demux/playlist/podcast.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/modules/demux/playlist/podcast.c b/modules/demux/playlist/podcast.c
index 42e730f052..5613156173 100644
--- a/modules/demux/playlist/podcast.c
+++ b/modules/demux/playlist/podcast.c
@@ -251,6 +251,7 @@ static int ReadDir( stream_t *p_demux, input_item_node_t *p_subitems )
                         input_item_AddInfo( p_current_input,
                             _( "Podcast Info" ), _( "Podcast Summary" ),
                             "%s", node );
+                        input_item_SetDescription( p_current_input, node );
                     }
                 }
                 else
-- 
2.50.1 (Apple Git-155)

