From 506bbbc13bd5290384c8e4a36d1b86d30ec518b4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <felix@feepk.net>
Date: Wed, 27 Nov 2019 20:57:31 +0100
Subject: [PATCH 23/24] apple/build: bootstrap only if configure does not exist

---
 extras/package/apple/build.sh | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/extras/package/apple/build.sh b/extras/package/apple/build.sh
index 8643f11d74..f2ceb2a971 100755
--- a/extras/package/apple/build.sh
+++ b/extras/package/apple/build.sh
@@ -585,7 +585,10 @@ fi
 
 # Bootstrap VLC
 cd "$VLC_SRC_DIR" || abort_err "Failed cd to VLC source dir"
-./bootstrap
+if ! [ -e configure ]; then
+    echo "Bootstraping vlc"
+    ./bootstrap
+fi
 
 # Build
 mkdir -p "${VLC_BUILD_DIR}/build/${VLC_PSEUDO_TRIPLET}"
-- 
2.21.0 (Apple Git-122.2)

