From 79201ae842cb6b16e7dc2fd188fcc75ec7ef0417 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <felix@feepk.net>
Date: Fri, 8 Apr 2022 17:31:01 +0200
Subject: [PATCH 23/24] appleOS build script: use bitcode support in contrib

---
 extras/package/apple/build.sh | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/extras/package/apple/build.sh b/extras/package/apple/build.sh
index 0322a03e3a..30364a560b 100755
--- a/extras/package/apple/build.sh
+++ b/extras/package/apple/build.sh
@@ -323,7 +323,7 @@ set_host_envvars()
     local bitcode_flag=""
     if [ "$VLC_USE_BITCODE" -gt "0" ]; then
         clike_flags+=" $VLC_BITCODE_FLAG"
-        bitcode_flag=" $VLC_BTICODE_FLAG"
+        bitcode_flag=" $VLC_BITCODE_FLAG"
     fi
 
     export CPPFLAGS="-arch $VLC_HOST_ARCH -isysroot $VLC_APPLE_SDK_PATH"
@@ -367,10 +367,6 @@ write_config_mak()
 {
     # Flags to be used for C-like compilers (C, C++, Obj-C)
     local clike_flags="$VLC_DEPLOYMENT_TARGET_CFLAG -arch $VLC_HOST_ARCH -isysroot $VLC_APPLE_SDK_PATH $1"
-    if [ "$VLC_USE_BITCODE" -gt "0" ]; then
-        # We use bitcode for contribs in every case, no dylib or executable built from them
-        clike_flags+=" -fembed-bitcode"
-    fi
 
     local vlc_cppflags="-arch $VLC_HOST_ARCH -isysroot $VLC_APPLE_SDK_PATH"
     local vlc_cflags="$clike_flags"
@@ -633,6 +629,11 @@ mkdir -p "$VLC_CONTRIB_INSTALL_DIR"
 # Set flag to error on partial availability
 write_config_mak "-Werror=partial-availability"
 
+if [ "$VLC_USE_BITCODE" -gt "0" ]; then
+    echo "using bitcode for contrib"
+    export BUILDWITHBITCODE="yes"
+fi
+
 # Bootstrap contribs
 ../bootstrap \
     --host="$VLC_HOST_TRIPLET" \
-- 
2.32.0 (Apple Git-132)

