From 7be4aca9d4e9c3cd3360421b1c54f43906814acf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Tue, 4 Aug 2026 15:45:08 +0200
Subject: [PATCH 25/27] ID3: read the TCMP compilation frame

---
 modules/meta_engine/ID3Meta.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/modules/meta_engine/ID3Meta.h b/modules/meta_engine/ID3Meta.h
index d82b88668a..69d5b94118 100644
--- a/modules/meta_engine/ID3Meta.h
+++ b/modules/meta_engine/ID3Meta.h
@@ -31,6 +31,7 @@ static struct
     const char *psz;
 } const ID3_tag_to_metatype[] = {
     { VLC_FOURCC('T', 'A', 'L', 'B'), vlc_meta_Album,       NULL },
+    { VLC_FOURCC('T', 'C', 'M', 'P'), vlc_meta_Compilation, NULL },
     { VLC_FOURCC('T', 'D', 'R', 'C'), vlc_meta_Date,        NULL },
     { VLC_FOURCC('T', 'E', 'N', 'C'), vlc_meta_extra,       "Encoder" },
     { VLC_FOURCC('T', 'I', 'T', '2'), vlc_meta_Title,       NULL },
-- 
2.50.1 (Apple Git-155)

