From ff83abe47fa0bb7544f08b756f31237a977cab75 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Tue, 4 Aug 2026 15:45:08 +0200
Subject: [PATCH 26/27] asf: read the WM/IsCompilation descriptor

---
 modules/demux/asf/asf.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/modules/demux/asf/asf.c b/modules/demux/asf/asf.c
index 5defd555dc..f370f9b66f 100644
--- a/modules/demux/asf/asf.c
+++ b/modules/demux/asf/asf.c
@@ -1395,6 +1395,7 @@ static int DemuxInit( demux_t *p_demux )
             else set_meta( "WM/AlbumArtist",  vlc_meta_AlbumArtist )
             else set_meta( "WM/Publisher",    vlc_meta_Publisher )
             else set_meta( "WM/PartOfSet",    vlc_meta_DiscNumber )
+            else set_meta( "WM/IsCompilation", vlc_meta_Compilation )
             else if( p_ecd->ppsz_value[i] != NULL && p_ecd->ppsz_name[i] &&
                     *p_ecd->ppsz_value[i] != '\0' && /* no empty value */
                     *p_ecd->ppsz_value[i] != '{'  && /* no guid value */
-- 
2.50.1 (Apple Git-155)

