From 8cb991fce30b3403f86c0546d35381536ee48f21 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Tue, 4 Aug 2026 15:47:08 +0200
Subject: [PATCH 27/27] taglib: read the compilation flag

---
 modules/meta_engine/taglib.cpp | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/modules/meta_engine/taglib.cpp b/modules/meta_engine/taglib.cpp
index a64ca58d7b..d1c6486ad5 100644
--- a/modules/meta_engine/taglib.cpp
+++ b/modules/meta_engine/taglib.cpp
@@ -449,6 +449,7 @@ static void ReadMetaFromAPE( APE::Tag* tag, demux_meta_t* p_demux_meta, vlc_meta
     SET( "LANGUAGE", Language );
     SET( "PUBLISHER", Publisher );
     SET( "MUSICBRAINZ_TRACKID", TrackID );
+    SET( "COMPILATION", Compilation );
 
     SET_EXTRA( "MUSICBRAINZ_ALBUMID", VLC_META_EXTRA_MB_ALBUMID );
 
@@ -510,6 +511,10 @@ static void ReadMetaFromASF( ASF::Tag* tag, demux_meta_t* p_demux_meta, vlc_meta
 #undef SET
 #undef SET_EXTRA
 
+    if( tag->attributeListMap().contains("WM/IsCompilation") )
+        vlc_meta_SetCompilation( p_meta,
+            tag->attributeListMap()["WM/IsCompilation"].front().toBool() ? "1" : "0" );
+
     // List the pictures
     list = tag->attributeListMap()["WM/Picture"];
     ASF::AttributeList::Iterator iter;
@@ -767,6 +772,7 @@ static void ReadMetaFromId3v2( ID3v2::Tag* tag, demux_meta_t* p_demux_meta, vlc_
     SET( "TLAN", Language );
     SET( "TPUB", Publisher );
     SET( "TPE2", AlbumArtist );
+    SET( "TCMP", Compilation );
     SET_EXTRA( "USLT", "Lyrics" );
 
 #undef SET_EXTRA
@@ -826,6 +832,7 @@ static void ReadMetaFromXiph( Ogg::XiphComment* tag, demux_meta_t* p_demux_meta,
     SET( "MUSICBRAINZ_TRACKID", TrackID );
     SET( "ALBUMARTIST", AlbumArtist );
     SET( "DISCNUMBER", DiscNumber );
+    SET( "COMPILATION", Compilation );
 
     SET_EXTRA( "MUSICBRAINZ_ALBUMID", VLC_META_EXTRA_MB_ALBUMID );
 #undef SET
@@ -885,6 +892,9 @@ static void ReadMetaFromMP4( MP4::Tag* tag, demux_meta_t *p_demux_meta, vlc_meta
 #undef SET
 #undef SET_EXTRA
 
+    if( tag->contains("cpil") )
+        vlc_meta_SetCompilation( p_meta, tag->item("cpil").toBool() ? "1" : "0" );
+
     if( tag->contains("covr") )
     {
         MP4::CoverArtList cover_list = tag->item("covr").toCoverArtList();
-- 
2.50.1 (Apple Git-155)

