From f840c7cbd8d479b4d203cc14f3a8618c9970cab2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=83=C2=BChne?= <fkuehne@videolan.org>
Date: Tue, 12 Jul 2016 13:50:54 +0200
Subject: [PATCH 07/17] contrib/avcodec: enforce use of FFmpeg

---
 contrib/src/ffmpeg/rules.mak | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/contrib/src/ffmpeg/rules.mak b/contrib/src/ffmpeg/rules.mak
index eabf605179..c37d463706 100644
--- a/contrib/src/ffmpeg/rules.mak
+++ b/contrib/src/ffmpeg/rules.mak
@@ -2,10 +2,10 @@
 
 #Uncomment the one you want
 #USE_LIBAV ?= 1
-#USE_FFMPEG ?= 1
+USE_FFMPEG ?= 1
 
 ifdef USE_FFMPEG
-FFMPEG_HASH=0768aaec1d683226e613e692080a588359c31334
+FFMPEG_HASH=911417f0b34e611bf084319c5b5a4e4e630da940
 FFMPEG_SNAPURL := http://git.videolan.org/?p=ffmpeg.git;a=snapshot;h=$(FFMPEG_HASH);sf=tgz
 FFMPEG_GITURL := http://git.videolan.org/git/ffmpeg.git
 else
-- 
2.11.0

