From 95792e45dec1bba78c8c07645d3cc8f82c8d82e4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Wed, 4 May 2016 16:00:40 +0200
Subject: [PATCH 08/20] libvlc: create VT-zero-copy variable

---
 lib/media_player.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/media_player.c b/lib/media_player.c
index 3f95c27c39..930190cd55 100644
--- a/lib/media_player.c
+++ b/lib/media_player.c
@@ -626,6 +626,7 @@ libvlc_media_player_new( libvlc_instance_t *instance )
 #endif
 #ifdef __APPLE__
     var_Create (mp, "drawable-nsobject", VLC_VAR_ADDRESS);
+    var_Create (mp, "videotoolbox-zero-copy", VLC_VAR_BOOL | VLC_VAR_DOINHERIT);
 #endif
 #ifdef __ANDROID__
     var_Create (mp, "drawable-androidwindow", VLC_VAR_ADDRESS);
-- 
2.11.0

