From d9610d5c0eb20d91c3bc7cb37fbdab9ae917c8bf Mon Sep 17 00:00:00 2001
From: Thomas Guillem <thomas@gllm.fr>
Date: Fri, 22 Jul 2016 11:20:24 +0200
Subject: [PATCH 14/20] taglib: DO NOT PUSH: works with networks accesses

---
 modules/meta_engine/taglib.cpp | 15 ++++-----------
 1 file changed, 4 insertions(+), 11 deletions(-)

diff --git a/modules/meta_engine/taglib.cpp b/modules/meta_engine/taglib.cpp
index 6d051ba1ea..ea3744da29 100644
--- a/modules/meta_engine/taglib.cpp
+++ b/modules/meta_engine/taglib.cpp
@@ -174,7 +174,7 @@ public:
 
     FileName name() const
     {
-        return m_stream->psz_location;
+        return m_stream->psz_url;
     }
 
     ByteVector readBlock(ulong length)
@@ -835,16 +835,8 @@ static int ReadMeta( vlc_object_t* p_this)
     if( unlikely(psz_uri == NULL) )
         return VLC_ENOMEM;
 
-    char *psz_path = vlc_uri2path( psz_uri );
-#if VLC_WINSTORE_APP && TAGLIB_VERSION >= TAGLIB_VERSION_1_11
-    if( psz_path == NULL )
-    {
-        free( psz_uri );
-        return VLC_EGENERIC;
-    }
-    free( psz_path );
-
-    stream_t *p_stream = vlc_access_NewMRL( p_this, psz_uri );
+#if TAGLIB_VERSION >= TAGLIB_VERSION_1_11
+    stream_t *p_stream = vlc_stream_NewMRL( p_this, psz_uri );
     free( psz_uri );
     if( p_stream == NULL )
         return VLC_EGENERIC;
@@ -852,6 +844,7 @@ static int ReadMeta( vlc_object_t* p_this)
     VlcIostream s( p_stream );
     f = FileRef( &s );
 #else /* VLC_WINSTORE_APP */
+    char *psz_path = vlc_uri2path( psz_uri );
     free( psz_uri );
     if( psz_path == NULL )
         return VLC_EGENERIC;
-- 
2.11.0

