From 1efb72d71dd4d9d94d9bd98072160149c9f07a46 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Mon, 17 Jul 2017 17:05:51 +0200
Subject: [PATCH 15/18] contrib/avcodec: enforce use of FFmpeg

---
 contrib/src/ffmpeg/rules.mak | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/contrib/src/ffmpeg/rules.mak b/contrib/src/ffmpeg/rules.mak
index 3e5f910e99..6acb5da740 100644
--- a/contrib/src/ffmpeg/rules.mak
+++ b/contrib/src/ffmpeg/rules.mak
@@ -2,7 +2,7 @@
 
 #Uncomment the one you want
 #USE_LIBAV ?= 1
-#USE_FFMPEG ?= 1
+USE_FFMPEG ?= 1
 
 ifdef USE_FFMPEG
 FFMPEG_HASH=a82468514048fb87d9bf38689866bc3b9aaccd02
-- 
2.14.1

