From eec11b618242a496b66e6a7512e7bc1bc8e26f91 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <felix@serioese.gmbh>
Date: Wed, 6 Sep 2017 17:38:35 +0200
Subject: [PATCH 18/18] vout iOS: don't draw if the app is not active

---
 modules/video_output/ios.m | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/modules/video_output/ios.m b/modules/video_output/ios.m
index a22d281b4d..9e59212a1e 100644
--- a/modules/video_output/ios.m
+++ b/modules/video_output/ios.m
@@ -529,7 +529,6 @@ static void OpenglESSwap(vlc_gl_t *gl)
 
 - (void)dealloc
 {
-
     [[NSNotificationCenter defaultCenter] removeObserver:self];
     [_eaglContext release];
     [super dealloc];
@@ -551,6 +550,10 @@ static void OpenglESSwap(vlc_gl_t *gl)
         return;
     }
 
+    if (unlikely(!_appActive)) {
+        return;
+    }
+
     glDisable(GL_DEPTH_TEST);
 
     glGenFramebuffers(1, &_frameBuffer);
-- 
2.14.1

