From d4b91031dfff129f8be625adbb8e735a44e2d889 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <felix@feepk.net>
Date: Sat, 20 Feb 2021 14:05:55 +0100
Subject: [PATCH 40/44] contrib/vpx: fix cross-compiling for macOS from macOS

---
 contrib/src/vpx/rules.mak | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/contrib/src/vpx/rules.mak b/contrib/src/vpx/rules.mak
index c8a146e9a9..8a805acefc 100644
--- a/contrib/src/vpx/rules.mak
+++ b/contrib/src/vpx/rules.mak
@@ -139,6 +139,9 @@ else
 VPX_LDFLAGS := -L$(IOS_SDK)/usr/lib -isysroot $(IOS_SDK) -miphoneos-version-min=8.4
 endif
 endif
+ifdef HAVE_MACOSX
+VPX_LDFLAGS := -L$(MACOSX_SDK)/usr/lib -isysroot $(MACOSX_SDK) -mmacosx-version-min=10.9
+endif
 VPX_LDFLAGS += -arch $(PLATFORM_SHORT_ARCH)
 endif
 
-- 
2.24.3 (Apple Git-128)

