From 742f85af09b16bb54f9ec9fcb3ad8af9825b8cec Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <felix@feepk.net>
Date: Wed, 10 Mar 2021 10:23:13 +0100
Subject: [PATCH 41/42] contrib/vpx: fix cross-compiling ARM-64 Simulator on
 x86_64

---
 contrib/src/vpx/rules.mak | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/contrib/src/vpx/rules.mak b/contrib/src/vpx/rules.mak
index 52d80a0b24..8366d98e69 100644
--- a/contrib/src/vpx/rules.mak
+++ b/contrib/src/vpx/rules.mak
@@ -134,11 +134,7 @@ ifdef HAVE_DARWIN_OS
 VPX_CONF += --enable-vp8-decoder --disable-tools
 VPX_CONF += --extra-cflags="$(CFLAGS) $(EXTRA_CFLAGS)"
 ifdef HAVE_IOS
-ifdef HAVE_TVOS
-VPX_LDFLAGS := -L$(IOS_SDK)/usr/lib -isysroot $(IOS_SDK) -mtvos-version-min=9.0
-else
-VPX_LDFLAGS := -L$(IOS_SDK)/usr/lib -isysroot $(IOS_SDK) -miphoneos-version-min=8.4
-endif
+VPX_LDFLAGS := -L$(IOS_SDK)/usr/lib -isysroot $(IOS_SDK) $(LDFLAGS)
 endif
 ifdef HAVE_MACOSX
 VPX_LDFLAGS := -L$(MACOSX_SDK)/usr/lib -isysroot $(MACOSX_SDK) -mmacosx-version-min=10.7
-- 
2.30.1 (Apple Git-130)

