From f829a4460ac542ed4c00ac64efbc7b4c9e57411c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <felix@feepk.net>
Date: Mon, 28 Jun 2021 13:11:15 +0200
Subject: [PATCH 42/42] http/cookie: fix double-free

---
 lib/media_player.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/lib/media_player.c b/lib/media_player.c
index 31ca3ed301..99d062a3e4 100644
--- a/lib/media_player.c
+++ b/lib/media_player.c
@@ -927,7 +927,6 @@ static void libvlc_media_player_destroy( libvlc_media_player_t *p_mi )
     if ( cookies )
     {
         var_Destroy( p_mi, "http-cookies" );
-        vlc_http_cookies_destroy( cookies );
     }
 
     libvlc_instance_t *instance = p_mi->p_libvlc_instance;
-- 
2.30.1 (Apple Git-130)

