From d6a09f4c9ab75fdd57465420e4203640f8ed9da6 Mon Sep 17 00:00:00 2001
From: Maxime Chapelet <umxprime@videolabs.io>
Date: Fri, 7 Jan 2022 11:34:24 +0100
Subject: [PATCH 45/46] videotoolbox : reload module if vout initialisation
 fails

---
 modules/codec/videotoolbox.m | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/modules/codec/videotoolbox.m b/modules/codec/videotoolbox.m
index 9e19b65d0f..54e070d5d4 100644
--- a/modules/codec/videotoolbox.m
+++ b/modules/codec/videotoolbox.m
@@ -105,6 +105,7 @@
     VTSESSION_STATUS_RESTART,
     VTSESSION_STATUS_RESTART_CHROMA,
     VTSESSION_STATUS_ABORT,
+    VTSESSION_STATUS_VOUT_FAILURE,
 };
 
 static int ConfigureVout(decoder_t *);
@@ -1937,6 +1938,10 @@ static int DecodeBlock(decoder_t *p_dec, block_t *p_block)
          * this ES */
         var_Create(p_dec, "videotoolbox-failed", VLC_VAR_VOID);
         return VLCDEC_RELOAD;
+    } else if (p_sys->vtsession_status == VTSESSION_STATUS_VOUT_FAILURE) 
+    {
+        vlc_mutex_unlock(&p_sys->lock);
+        return VLCDEC_RELOAD;
     }
     vlc_mutex_unlock(&p_sys->lock);
 
@@ -2107,7 +2112,7 @@ static int UpdateVideoFormat(decoder_t *p_dec, CVPixelBufferRef imageBuffer)
     p_dec->fmt_out.video.i_chroma = p_dec->fmt_out.i_codec;
     if (decoder_UpdateVideoFormat(p_dec) != 0)
     {
-        p_dec->p_sys->vtsession_status = VTSESSION_STATUS_ABORT;
+        p_dec->p_sys->vtsession_status = VTSESSION_STATUS_VOUT_FAILURE;
         return -1;
     }
     return 0;
-- 
2.32.0 (Apple Git-132)

