From e639a4efd525a34e84eebf5f58aae7cd495ef889 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <felix@feepk.net>
Date: Tue, 22 Mar 2022 12:32:47 +0100
Subject: [PATCH 46/52] contrib/upnp: disable IPv6 support on iOS

---
 contrib/src/upnp/rules.mak | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/contrib/src/upnp/rules.mak b/contrib/src/upnp/rules.mak
index 8f0f38570d..972a6860da 100644
--- a/contrib/src/upnp/rules.mak
+++ b/contrib/src/upnp/rules.mak
@@ -20,8 +20,12 @@ endif
 ifdef HAVE_WINSTORE
 CONFIGURE_ARGS=--disable-ipv6 --enable-unspecified_server
 else
+ifdef HAVE_IOS
+CONFIGURE_ARGS=--disable-ipv6 --disable-webserver
+else
 CONFIGURE_ARGS=--enable-ipv6
 endif
+endif
 ifndef WITH_OPTIMIZATION
 CONFIGURE_ARGS += --enable-debug
 endif
-- 
2.32.0 (Apple Git-132)

