From 616b4ffb5169bf96cebe0506a794caf84cb4e3f0 Mon Sep 17 00:00:00 2001
From: Thomas Guillem <thomas@gllm.fr>
Date: Fri, 7 Jan 2022 14:28:20 +0100
Subject: [PATCH 54/56] dsm: make netbios_ns interruptible

---
 modules/access/dsm/access.c | 24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/modules/access/dsm/access.c b/modules/access/dsm/access.c
index e3a0dd12bf..167396bf7b 100644
--- a/modules/access/dsm/access.c
+++ b/modules/access/dsm/access.c
@@ -167,10 +167,30 @@ smb_session_interrupt_unregister( void )
     vlc_interrupt_unregister();
 }
 
+static void
+netbios_ns_interrupt_callback( void *data )
+{
+    netbios_ns_abort( data );
+}
+
+static inline void
+netbios_ns_interrupt_register( netbios_ns *ns )
+{
+    vlc_interrupt_register( netbios_ns_interrupt_callback, ns );
+}
+
+static inline void
+netbios_ns_interrupt_unregister( void )
+{
+    vlc_interrupt_unregister();
+}
+
 #else
 
 #define smb_session_interrupt_register( sys ) do {} while (0)
 #define smb_session_interrupt_unregister() do {} while(0)
+#define netbios_ns_interrupt_register( ns ) do {} while (0)
+#define netbios_ns_interrupt_unregister() do {} while (0)
 
 #endif
 
@@ -311,6 +331,7 @@ static int get_address( stream_t *p_access )
         netbios_ns *p_ns = netbios_ns_new();
         if( p_ns == NULL )
             return VLC_EGENERIC;
+        netbios_ns_interrupt_register( p_ns );
 
         /* Is this a netbios name on this LAN ? */
         uint32_t ip4_addr;
@@ -318,6 +339,7 @@ static int get_address( stream_t *p_access )
         int ret = netbios_ns_resolve( p_ns, p_sys->url.psz_host,
                                       NETBIOS_FILESERVER,
                                       &p_sys->addr.s_addr);
+        netbios_ns_interrupt_unregister();
         netbios_ns_destroy( p_ns );
 
         if( ret == 0 )
@@ -347,10 +369,12 @@ static int get_address( stream_t *p_access )
     netbios_ns *p_ns = netbios_ns_new();
     if( p_ns == NULL )
         return VLC_EGENERIC;
+    netbios_ns_interrupt_register( p_ns );
 
     /* We have an IP address, let's find the NETBIOS name */
     const char *psz_nbt = netbios_ns_inverse( p_ns, p_sys->addr.s_addr );
 
+    netbios_ns_interrupt_unregister();
     netbios_ns_destroy( p_ns );
 
     if( psz_nbt != NULL )
-- 
2.32.0 (Apple Git-132)

