From a1b9e1bf336b7e30babdd62b97328db73831fc58 Mon Sep 17 00:00:00 2001
From: Thomas Guillem <thomas@gllm.fr>
Date: Mon, 11 Apr 2022 14:34:30 +0200
Subject: [PATCH 71/80] smb2: homogenize status and error functions

---
 modules/access/smb2.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/modules/access/smb2.c b/modules/access/smb2.c
index 3032627505..c2b7c6d5d7 100644
--- a/modules/access/smb2.c
+++ b/modules/access/smb2.c
@@ -155,7 +155,7 @@ vlc_smb2_op_reset(struct vlc_smb2_op *op, struct smb2_context **smb2p)
 }
 
 static int
-smb2_check_status(struct vlc_smb2_op *op, int status, const char *psz_func)
+smb2_check_status(struct vlc_smb2_op *op, const char *psz_func, int status)
 {
     if (status < 0)
     {
@@ -188,7 +188,7 @@ smb2_set_error(struct vlc_smb2_op *op, const char *psz_func, int err)
 }
 
 #define VLC_SMB2_CHECK_STATUS(op, status) \
-    smb2_check_status(op, status, __func__)
+    smb2_check_status(op, __func__, status)
 
 #define VLC_SMB2_SET_ERROR(op, func, err) \
     smb2_set_error(op, func, err)
-- 
2.32.0 (Apple Git-132)

