From 51d2d2426d10e5725093c1eb0548eefac049b21b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <felix@feepk.net>
Date: Fri, 8 Apr 2022 14:13:52 +0200
Subject: [PATCH 72/72] smb2: 32-bit compilation fix

---
 modules/access/smb2.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/modules/access/smb2.c b/modules/access/smb2.c
index 3032627505..adb981039d 100644
--- a/modules/access/smb2.c
+++ b/modules/access/smb2.c
@@ -208,7 +208,11 @@ __attribute__((weak)) const t_socket *
 smb2_get_fds(struct smb2_context *smb2, size_t *fd_count, int *timeout)
 {
     (void) timeout;
+#if defined (__i386__) || defined (__arm__)
+    static t_socket fd;
+#else
     static thread_local t_socket fd;
+#endif
 
     *fd_count = 1;
     fd = smb2_get_fd(smb2);
-- 
2.32.0 (Apple Git-132)

