From 1cdd61f3688f96480caf1ee8497d6304689707e0 Mon Sep 17 00:00:00 2001
From: Thomas Guillem <thomas@gllm.fr>
Date: Thu, 14 Apr 2022 14:05:48 +0200
Subject: [PATCH 73/77] smb2: rework error handling from smb2_service_fd

---
 modules/access/smb2.c | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/modules/access/smb2.c b/modules/access/smb2.c
index c2b7c6d5d7..0902ada7b8 100644
--- a/modules/access/smb2.c
+++ b/modules/access/smb2.c
@@ -178,10 +178,9 @@ smb2_set_error(struct vlc_smb2_op *op, const char *psz_func, int err)
     if (op->log && err != -EINTR)
         msg_Err(op->log, "%s failed: %d, %s", psz_func, err, smb2_get_error(op->smb2));
 
-    if (err != 0)
+    /* Don't override if set via smb2_check_status */
+    if (op->error_status == 0)
         op->error_status = err;
-    else if (op->error_status == 0) /* don't override if set via smb2_check_status */
-        op->error_status = -EINVAL;
 
     smb2_destroy_context(op->smb2);
     *op->smb2p = NULL;
@@ -249,7 +248,7 @@ vlc_smb2_mainloop(struct vlc_smb2_op *op)
         else if (ret == 0)
         {
             if (smb2_service_fd(op->smb2, -1, 0) < 0)
-                VLC_SMB2_SET_ERROR(op, "smb2_service", 0);
+                VLC_SMB2_SET_ERROR(op, "smb2_service", -EINVAL);
         }
         else
         {
@@ -257,7 +256,7 @@ vlc_smb2_mainloop(struct vlc_smb2_op *op)
             {
                 if (p_fds[i].revents
                  && smb2_service_fd(op->smb2, p_fds[i].fd, p_fds[i].revents) < 0)
-                    VLC_SMB2_SET_ERROR(op, "smb2_service", 0);
+                    VLC_SMB2_SET_ERROR(op, "smb2_service", -EINVAL);
             }
         }
     }
-- 
2.32.0 (Apple Git-132)

