From c9cdbc656476194b0dd212f11265fb3daf548d69 Mon Sep 17 00:00:00 2001
From: Thomas Guillem <thomas@gllm.fr>
Date: Mon, 11 Apr 2022 14:36:10 +0200
Subject: [PATCH 73/76] smb2: move smb2_set_error

To avoid forward declarations in the next commit.
---
 modules/access/smb2.c | 34 +++++++++++++++++-----------------
 1 file changed, 17 insertions(+), 17 deletions(-)

diff --git a/modules/access/smb2.c b/modules/access/smb2.c
index c2b7c6d5d7..3038747773 100644
--- a/modules/access/smb2.c
+++ b/modules/access/smb2.c
@@ -154,6 +154,21 @@ vlc_smb2_op_reset(struct vlc_smb2_op *op, struct smb2_context **smb2p)
     op->error_status = 0;
 }
 
+static void
+smb2_set_error(struct vlc_smb2_op *op, const char *psz_func, int err)
+{
+    if (op->log && err != -EINTR)
+        vlc_error(op->log, "%s failed: %d, %s", psz_func, err, smb2_get_error(op->smb2));
+
+    if (err != 0)
+        op->error_status = err;
+    else if (op->error_status == 0) /* don't override if set via smb2_check_status */
+        op->error_status = -EINVAL;
+
+    smb2_destroy_context(op->smb2);
+    *op->smb2p = NULL;
+}
+
 static int
 smb2_check_status(struct vlc_smb2_op *op, const char *psz_func, int status)
 {
@@ -172,27 +187,12 @@ smb2_check_status(struct vlc_smb2_op *op, const char *psz_func, int status)
     }
 }
 
-static void
-smb2_set_error(struct vlc_smb2_op *op, const char *psz_func, int err)
-{
-    if (op->log && err != -EINTR)
-        msg_Err(op->log, "%s failed: %d, %s", psz_func, err, smb2_get_error(op->smb2));
-
-    if (err != 0)
-        op->error_status = err;
-    else if (op->error_status == 0) /* don't override if set via smb2_check_status */
-        op->error_status = -EINVAL;
-
-    smb2_destroy_context(op->smb2);
-    *op->smb2p = NULL;
-}
+#define VLC_SMB2_SET_ERROR(op, func, err) \
+    smb2_set_error(op, func, err)
 
 #define VLC_SMB2_CHECK_STATUS(op, status) \
     smb2_check_status(op, __func__, status)
 
-#define VLC_SMB2_SET_ERROR(op, func, err) \
-    smb2_set_error(op, func, err)
-
 #define VLC_SMB2_STATUS_DENIED(x) (x == -ECONNREFUSED || x == -EACCES)
 
 #if defined (__ELF__) || defined (__MACH__) /* weak support */
-- 
2.32.0 (Apple Git-132)

