From bcbdf3b80faa04fd1eccc33a20953888d1198cf1 Mon Sep 17 00:00:00 2001
From: Thomas Guillem <thomas@gllm.fr>
Date: Thu, 21 Apr 2022 08:17:44 +0200
Subject: [PATCH 78/80] smb2: rename logged seek function name

---
 modules/access/smb2.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/access/smb2.c b/modules/access/smb2.c
index ff41707a63..a8e71ad528 100644
--- a/modules/access/smb2.c
+++ b/modules/access/smb2.c
@@ -318,7 +318,7 @@ FileSeek(stream_t *access, uint64_t i_pos)
     int err = smb2_lseek(op.smb2, sys->smb2fh, i_pos, SEEK_SET, NULL);
     if (err < 0)
     {
-        VLC_SMB2_SET_ERROR(&op, "smb2_seek_async", err);
+        VLC_SMB2_SET_ERROR(&op, "smb2_lseek", err);
         return err;
     }
     sys->eof = false;
-- 
2.32.0 (Apple Git-132)

