From c81f76d0b2cccebe09096cbd21911c3a87722197 Mon Sep 17 00:00:00 2001
From: Thomas Guillem <thomas@gllm.fr>
Date: Thu, 21 Apr 2022 08:25:56 +0200
Subject: [PATCH 80/80] smb2: don't seek past INT64_MAX

---
 modules/access/smb2.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/modules/access/smb2.c b/modules/access/smb2.c
index 3065d10120..178e30909e 100644
--- a/modules/access/smb2.c
+++ b/modules/access/smb2.c
@@ -313,6 +313,13 @@ FileSeek(stream_t *access, uint64_t i_pos)
     if (sys->smb2 == NULL)
         return VLC_EGENERIC;
 
+    if (i_pos > INT64_MAX)
+    {
+        msg_Err(access, "can't seek past INT64_MAX (requested: %"PRIu64")\n",
+                i_pos);
+        return VLC_EGENERIC;
+    }
+
     struct vlc_smb2_op op = VLC_SMB2_OP(access, &sys->smb2);
 
     int64_t err = smb2_lseek(op.smb2, sys->smb2fh, i_pos, SEEK_SET, NULL);
-- 
2.32.0 (Apple Git-132)

